### DISCUSSION TITLE
a not optimized number row layout

### DISCUSSION THREAD

**Discussion by lpradopostigo** [binarybottle/engram-layout#3] [Ideas]
Hi, i was using this layout for 1 month (i came from dvorak) was hard at the start but i think it worth. I think the most notable difference is the reduction of stretch on the index finger, i have RSI and the overall pain reduction was notorious. 
I'm a programmer btw, so i didn't like the idea of trying to preserve the number row. Here is my suggestion based on some xah post for a number row, i didn't measure anything so i think there is room for improvement.
| left hand      | right hand |
| ----------- | ----------- |
|~ 9   7   5   1  3   /     | = 2 0 4 6 8     |
| ^ %  \| \    >  < {   |  }  & * [ ] +  |

**Comment by binarybottle** [comment]
Thank you very much for giving it a concerted go.  I am so happy that you are finding that the Engram layout is helping with your repetitive strain injury.

I am working on a revision of the Engram layout based on feedback I have received and experiments related to finger strength, so new version may be forthcoming.

I appreciate your suggestion for the number row, but I think I'm going to leave the numbers in the same order to make it easier to remember.  I also want the shift characters for the number keys to be easier to remember as well, so I'm using a different scheme than I would were I simply interested in frequency of occurrence.

**Reply by lpradopostigo** [reply]
I see, well i checked the table that i posted and i made a mistake. The layout was based on my [actual keyboard](https://pisces.bbystatic.com/image2/BestBuy_US/images/products/6379/6379631_sd.jpg) physical layout, just something to point and thanks for the layout i will try the new version.

Aside of that, i know that is your project , but there is something bugging me, why apply a different scheme on the number row? I mean its comfortable write near the home row, but writing symbols on the number row is kinda painful even using my layout (probably because the need of the shift state), maybe its better permute the state (symbols first, numbers second) i tried that, and is painful when you write CSS for example. I doubt that i can find a general case layout, but since you are measuring those things, i want to you consider this case.

**Comment by binarybottle** [comment]
Thank you again, and I appreciate your feedback. There are variants of layouts that intend to aid programmers, where the numbers and symbols are swapped, but this makes it difficult to write numbers unless you use a dedicated number pad.  I highly recommend using a keyboard like the ergodox, where a slight tap on a key will change to a shift state for one or more following keys, so you do not have to press the shift key while pressing another key.

**Reply by lpradopostigo** [reply]
Really interesting stuff from ergodox, for now i will keep using sticky keys feature. On the other side i realized that is painful to keep my hands on the keyboard; lets say some modal editor like vim makes you easy to reach most used actions, this seems great at first glance, but i tend to develop a habit to keep my hands attached to those keys, even when i am not writing. So i tried qwerty but instead of using the traditional touch type movements i move my whole hand up and down, left and right, no stretching. The pain was almost zero, i thought: typing habits really matters, at least for me. Now i using this approach to type using your layout, what do you think about it?

**Comment by binarybottle** [comment]
I have had the same experience.  When I am using a layout that requires lateral tendon movements, I also move my whole hand rather than stretch my fingers in uncomfortable ways.
________________________________

**Comment by binarybottle** [comment]
Please see https://github.com/binarybottle/engram-2021/discussions/86#discussion-10126586 for an announcement of the new Engram approach/layouts/repo!
